### PR TITLE
Log versions of Cylc, Rose and CylcRose in the rose-suite-cylc-instal…

### LINKED ISSUE #318: log cylc-rose version on installation
Would be good to record the cylc-rose version somewhere in the installation for debugging.

Ideas:
* `opt/rose-suite-cylc-install.conf`
* `log/install/?`

Would be a good idea to stick the metomi-rose version there too.

### PR DESCRIPTION
**Pull request by wxtim**
…l.conf

Closes #318 

Also took the opportunity to move `tests/unit/test_functional_post_install.py` to `tests/functional/test_post_install.py`. This has been bugging me for ages, but I dodn't want to muddy the waters of a more complex PR.

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` (and `conda-environment.yml` if present).
- [x] Tests are included (or explain why tests are not needed).
- [x] `CHANGES.md` not changed - this is a logging change only.
- [x] ~[Cylc-Doc](https://github.com/cylc/cylc-doc) pull request opened if required at cylc/cylc-doc/pull/XXXX.~
- [x] If this is a bug fix, PR should be raised against the relevant `?.?.x` branch.

### CODE REVIEW

File: cylc/rose/utilities.py [hunk 1]
```diff
@@ -957,6 +959,11 @@ def record_cylc_install_options(
             ]
 
     cli_config.comments = [' This file records CLI Options.']
+    cli_config.comments += [
+        ' Installed with:',
+        f'     * Cylc Rose: {CYLC_ROSE_VERSION}',
+        f'     * Rose     : {ROSE_VERSION}',
+        f'     * Cylc     : {CYLC_VERSION}']
```
Comment (oliver-sanders) [inline cylc/rose/utilities.py:966]:
```suggestion
    cli_config.comments = [
        ' This file records CLI Options.',
        ' Installed with:',
        f'     * Cylc Rose: {CYLC_ROSE_VERSION}',
        f'     * Rose     : {ROSE_VERSION}',
        f'     * Cylc     : {CYLC_VERSION}']
```

File: tests/functional/test_reinstall.py [hunk 2]
```diff
@@ -149,7 +150,7 @@ def test_cylc_install_files(fixture_install_flow, file_, expect):
         ),
         (
             'opt/rose-suite-cylc-install.conf', (
-                '# This file records CLI Options.\n\n'
+                '# This file records CLI Options.{version_info}\n'
```
Comment (oliver-sanders) [inline tests/functional/test_reinstall.py:153]: This isn't a format string.

The test still passes, so the `version_info` is presumably not require here?

File: tests/functional/test_reinstall.py [hunk 3]
```diff
@@ -190,7 +193,7 @@ async def test_cylc_reinstall_files(
...
```
Comment (oliver-sanders) [inline tests/functional/test_reinstall.py:196]: This isn't a format string.

The test still passes, so the `version_info` is presumably not require here?

File: tests/functional/test_reinstall_clean.py [hunk 4]
```diff
@@ -105,7 +105,7 @@ async def fixture_install_flow(
     [
         (
             'opt/rose-suite-cylc-install.conf', (
-                '# This file records CLI Options.\n\n'
+                '# This file records CLI Options.{version_info}\n'
```
Comment (oliver-sanders) [inline tests/functional/test_reinstall_clean.py:108]: This isn't a format string.

The test still passes, so the `version_info` is presumably not require here?

File: tests/functional/test_reinstall_clean.py [hunk 5]
```diff
@@ -125,7 +126,7 @@ def test_cylc_install_files(fixture_install_flow, file_, expect):
...
```
Comment (oliver-sanders) [inline tests/functional/test_reinstall_clean.py:129]: This isn't a format string.

The test still passes, so the `version_info` is presumably not require here?

File: tests/functional/test_reinstall.py (hunk 2 above)
Comment (wxtim) [reply tests/functional/test_reinstall.py:153]: No, it's not a f-string, but I use format on it later to insert `version_info`:

```python
expect = expect.format(version_info=version_info)
```

**Comment by oliver-sanders** [comment]
(lint failing)

**Review by oliver-sanders — APPROVED** [review]

**Review by markgrahamdawson — APPROVED** [review]

**Comment by markgrahamdawson** [comment]
I wont merge this until tests are passing on master?

**Review by markgrahamdawson — APPROVED** [review]

**Pull request merged**